qcodes/instrument_drivers/tektronix/keithley_7510.py

```diff
@@ -1,10 +1,372 @@
+        selected elements.
+        """
+        try:
+            _ = self.setpoints()
+        except NotImplementedError:
+            # if the "setpionts" has not been implemented, use a time series
+            # with parameters "t_start" and "t_stop":
+            self.set_setpoints(self.t_start, self.t_stop)
+
+        if self.parent.digi_sense_function() == "None":
+            # when current sense is not digitize sense
+            sense_function = self.parent.sense_function()
+            unit = Sense7510.function_modes[sense_function]["unit"]
+        else:
+            # when current sense is digitize sense
+            sense_function = self.parent.digi_sense_function()
+            unit = DigitizeSense7510.function_modes[sense_function]["unit"]
+
+        elements_units = {
+            "date": "str",
+            "measurement_formatted": "str",
+            "fractional_seconds": "s",
+            "measurement": unit,
+            "relative_time": "s",
+            "seconds": "s",
+            "measurement_status": "",
+            "time": "str",
+            "timestamp": "str",
+            "measurement_unit": "str"
+        }
+
+        if not self.elements():
+            raw_data = self.ask(f":TRACe:DATA? "
+                                f"{self.data_start()}, "
+                                f"{self.data_end()}, "
+                                f"'{self.short_name}'")
+        else:
+            elements = \
+                [self.buffer_elements[element] for element in self.elements()]
+            raw_data = self.ask(f":TRACe:DATA? {self.data_start()}, "
+                                f"{self.data_end()}, "
+                                f"'{self.short_name}', "
+                                f"{','.join(elements)}")
+
+        all_data = raw_data.split(",")
+
+        if len(self.elements()) == 0:
+            elements = ['measurement']
+        else:
+            elements = self.elements()
+        n_elements = len(elements)
+
+        units = tuple(elements_units[element] for element in elements)
+        processed_data = dict.fromkeys(elements)
+        for i, (element, unit) in enumerate(zip(elements, units)):
+            if unit == 'str':
+                processed_data[element] = all_data[i::n_elements]
+            else:
+                processed_data[element] = [
+                    float(v) for v in all_data[i::n_elements]
+                ]
+
+        data = DataArray7510(
+            names=tuple(elements),
+            shapes=((self.n_pts(),),) * n_elements,
+            units=units,
+            setpoints=((self.setpoints(),),) * n_elements,
+            setpoint_units=((self.setpoints.unit,),) * n_elements,
+        )
+        data._data = tuple(
+            tuple(processed_data[element]) for element in elements
+        )
+        for i in range(len(data.names)):
+            setattr(data, data.names[i], processed_data[data.names[i]])
```

Comment:
Why do we need this?

Reply:
To be honest, we want to access the data in the following way:
>data = buffer.data()
buffer.elements(['element1', 'element2']
data1 = data.element1   <---- this
data2 = data.element2    <---- and this. By doing so, the python lint would be able to check if "element1" is an attribute of object "data".

In addition to the QCoDeS way:
>dataset = datasaver.dataset
data = dataset.get_parameter_data()
data1 = data['element1']
data2 = data['element2']